apps/prairielearn/elements/pl-image-capture/pl-image-capture.py

```
@@ -0,0 +1,78 @@
+import chevron
+import lxml.html
+import prairielearn as pl
+
+
+def prepare(element_html: str, data: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+
+    pl.check_attribs(
+        element,
+        required_attribs=["answer-name"],
+        optional_attribs=[],
+    )
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
+    element = lxml.html.fragment_fromstring(element_html)
+
+    answer_name = pl.get_string_attrib(element, "answer-name", "1")
+
+    if data["panel"] != "question":
+        return ""
+
+    submitted_files = data["submitted_answers"].get("_files", [])
+
+    submitted_file_name = None
+    if len(submitted_files) > 0:
+        submitted_file_name = submitted_files[0].get("name")
+
+    html_params = {
+        "answer_name": answer_name,
+        "variant_id": data["options"].get("variant_id", ""),
+        "submitted_file_name": submitted_file_name,
+        "submission_date": data["options"].get("submission_date", ""),
+        "submission_files_url": data["options"].get("submission_files_url", None),
+        "uuid": pl.get_uuid(),
+    }
+
+    course_id = data["options"].get("course_id", None)
+    course_instance_id = data["options"].get("course_instance_id", None)
+    question_id = data["options"].get("question_id", None)
+    instance_question_id = data["options"].get("instance_question_id", None)
+
+    if course_instance_id is not None and instance_question_id is not None:
+        external_image_capture_url = f"{data['options'].get('serverCanonicalHost')}/pl/course_instance/{course_instance_id}/instance_question/{instance_question_id}/variants/{html_params['variant_id']}/external_image_capture/answer_name/{html_params['answer_name']}"
```

I think you should use the normal parenthesis lookup for `'serverCanonicalHost'`, since if this not present, the lookup will return 'None' and produce an invalid URL, so it would be better to just throw an exception rather than produce a strange URL with errors.

---

Got it; thank you for the heads up on this, @eliotwrobson . I'll make that change 👍

---

I made this change:

https://github.com/PrairieLearn/PrairieLearn/blob/d0044d9611d68497d3587fdf6ac0249d36e2fd17/apps/prairielearn/elements/pl-image-capture/pl-image-capture.py#L57-L73